cli/dcoscli/task/main.py

```diff
@@ -161,6 +173,167 @@ def _log(follow, completed, lines, task, file_):
+        logger.debug('Full task state: {}'.format(task))
+        raise DCOSException('Invalid executor info. '
+                            'Missing field `state`')
+
+    container_ids = []
+    statuses = task.get('statuses')
+    if not statuses:
+        logger.debug('Full task state: {}'.format(task))
+        raise DCOSException('Invalid executor info. Missing field `statuses`')
+
+    for status in statuses:
+        if 'state' not in status:
+            logger.debug('Full task state: {}'.format(task))
+            raise DCOSException('Invalid executor info. Missing field `state`')
+
+        if status['state'] != task_state:
+            continue
+
+        container_status = status.get('container_status')
+        if not container_status:
+            logger.debug('Full task state: {}'.format(task))
+            raise DCOSException('Invalid executor info. '
+                                'Missing field `container_status`')
+
+        container_id = container_status.get('container_id')
+        if not container_id:
+            logger.debug('Full task state: {}'.format(task))
+            raise DCOSException('Invalid executor info. '
+                                'Missing field `container_id`')
+
+        # traverse nested container_id field
+        while True:
+            value = container_id.get('value')
+            if not value:
+                logger.debug('Full task state: {}'.format(task))
+                raise DCOSException('Invalid executor info. Missing field'
+                                    '`value` for nested container ids')
+
+            container_ids.append(value)
+
+            if 'parent' not in container_id:
+                break
+
+            container_id = container_id['parent']
+
+    return '.'.join(reversed(container_ids))
+
+
+def _dcos_log(follow, tasks, lines, file_, completed):
+    """ a client to dcos-log
+
+    :param follow: same as unix tail's -f
+    :type follow: bool
+    :param task: task pattern to match
+    :type task: str
+    :param lines: number of lines to print
+    :type lines: int
+    :param file_: file path to read
+    :type file_: str
+    :param completed: whether to include completed tasks
+    :type completed: bool
+    """
+
+    # only stdout and stderr is supported
+    if file_ not in ('stdout', 'stderr'):
+        raise DCOSException('Expect file stdout or stderr. '
+                            'Got {}'.format(file_))
+    # state json may container tasks and completed_tasks fields. Based on
+    # user request we should traverse the appropriate field.
+    tasks_field = 'tasks'
+    if completed:
+        tasks_field = 'completed_tasks'
+
+    for task in tasks:
+        executor_info = task.executor()
+        if (tasks_field not in executor_info and
+                not isinstance(executor_info[tasks_field], list)):
+            logger.debug('Executor info: {}'.format(executor_info))
+            raise DCOSException('Invalid executor info. '
```

Comment:
since we're catching all DCOSExceptions, this error will never be seen. instead of printing the error here: https://github.com/dcos/dcos-cli/pull/817/files#diff-2e59d2980b0b9551474a304b82f61bd3R494, lets add it to the debug output so we know where we failed.